app/jobs/update_warehouse_clients_caches_job.rb

```diff
@@ -0,0 +1,39 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class UpdateWarehouseClientsCachesJob < BaseJob
+  queue_as ENV.fetch('DJ_LONG_QUEUE_NAME', :long_running)
+  WAIT_MINUTES = 4
+
+  def perform(client_ids: [], include_cas_and_cohorts: false, skip_expensive_calculations: false)
+    # If any for this class are already running, requeue for a few minutes in the future
+    any_jobs_of_type_other_than_self = Delayed::Job.jobs_for_class(self.class.name).
+      running.
+      where.not(Delayed::Job.arel_table[:handler].matches("%#{job_id}%")).
+      exists?
+    return requeue_job if any_jobs_of_type_other_than_self
```

Comment:
@ttoomey this is the pattern we've used before, but I suppose this could just take out an advisory lock with the class name and then each run could attempt to open the lock, if it fails, it could re-queue.  Do you have a preference for pattern?